src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -302,32 +300,84 @@ export class EditorComponent implements OnInit {
     }
     this.drawMask();
+    if (this.maskTool == MaskTool.magicWand) {
+      this.disableFloodFill = false;
+    }
     this.disableSubmit = false;
-    if (this.maskTool == 'magic-wand') {
-      this.disableFloodFill = false
+  }
+
+
+  /**  Retrieves new tolerance value from child component toolbar and updates. */
+  updateTolerance(value: number) {
+    this.tolerance = value;
+    console.log('new tolerance: ' + value);
+  }
+
+
+ /** 
+  *  Retrieves new tint value from child component toolbar and draws mask with new tint.
+  *  If the tint value cannot be larger than 1 and if the value inputed is greater than 1, 
+  *    then the tint defaults to 1.
+  */
+  updateTint(value: number) {
+    if (value > 1) {
+      this.tint = 1;
+    }
+    else {
+      this.tint = value;
+    }
+    //  Draw mask with new tint value.
+    this.disableFloodFill = true;
+    this.drawMask();
+    if (this.maskTool == MaskTool.magicWand) {
+      this.disableFloodFill = false;
     }
+    console.log('new tint: ' + value);
   }
 
  /** 
   *  Updates the value of the Toolbar toggle group.
-  *  All cases beside 'magic-wand' must disableFloodFill.
   */
   updateMaskTool(tool: string) {
-    // change value selected on form;
-    this.maskTool = tool;
+    console.log('New Tool: ' + tool);
+    //  All cases beside 'magic-wand' must disableFloodFill.
     this.disableFloodFill = true;
     switch (tool) {
-      case 'magic-wand': 
+      case 'MAGIC-WAND': 
+        this.maskTool = MaskTool.magicWand;
         this.disableFloodFill = false;
-        // redraw mask and image
-        this.drawMask();
         break;
-      case 'mask-only':
-        let mask = new Image();
-        this.disableSubmit = true;
-        this.drawMask();
-        this.disableSubmit = false;
+      case 'PAINT':
+        this.maskTool = MaskTool.paint;
+        break;
+      case 'MASK-ONLY':
+        this.maskTool = MaskTool.maskOnly;
         break;
     }
+    console.log('switched tool to ' + this.maskTool);
+    //  Always redraw mask/image when switching between features because of MaskOnly tool.
+    this.drawMask();
+  }
+
+ /**
+  *  Adds pixel user painted to mask.
+  *  TODO: Possibly change the implementation so the pixel drawn is
+  *        replecated on the screen ASAP, and then once the user finishes 
+  *        drawing (mouse up) then the real mask is drawn based on the set.
+  *        Would decrease lag.
+  */
+  drawPixel(pixel: number) {
+    //Commented out would be the TODO but the pixels don't line up 1:1 so there would be a visual difference in what the 
+    /// user thinks they're clicking on and what they actully paint on the mask.
+    // let imageData = this.ctx.getImageData(0,0,this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+    // imageData.data[pixel] = 255;
+    // imageData.data[pixel + 2] = 255;
+    // imageData.data[pixel + 3] = 255;
+    // this.ctx.putImageData(imageData, 0,0);
```

Comment:
Can this commented out code be removed?  The following seems to do the same thing and it reuses the drawMask function instead of doing it on its own.